Add tests

PratikMane0112 opened the pull request:
- closes #19

Created a comprehensive test suite for the Additional Identities plugin based on the provided references:
Basic unit tests for all classes:
- AdditionalIdentityTest.java - Tests the basic functionality of AdditionalIdentity class
- AdditionalIdentitiesTest.java - Tests the AdditionalIdentities user property
Integration tests that use JenkinsRule:
- AdditionalIdentityResolverTest.java - Basic tests for the identity resolver
- AdditionalIdentitiesIntegrationTest.java - Comprehensive tests that verify the entire plugin functionality

### Testing done

Tested locally with `mvn clean install`

### Submitter checklist
- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [X] Link to relevant issues in GitHub or Jira
- [X] Ensure you have provided tests - that demonstrates feature works or fixes the issue

Please suggest any changes or updates, if needed

viceice commented on src/test/java/com/cloudbees/jenkins/plugins/AdditionalIdentitiesIntegrationTest.java:
```diff
@@ -0,0 +1,145 @@
+        alice.addProperty(new AdditionalIdentities(aliceIdentities));
+
+        // Configure Bob with Bitbucket and JIRA identities
+        List<AdditionalIdentity> bobIdentities = new ArrayList<>();
```
use `var` instead of type where possible

PratikMane0112 commented:
Sure @viceice , I will make updates accordingly in near future days.

PratikMane0112 commented:
- Changed assertions to use JUnit 5 style with messages as the last parameter
- Used the JUnit 5 testing approach while maintaining JUnit 4 compatibility where needed for Jenkins test harness
- Updated variable declarations to use var where possible

@viceice Also suggest any changes needs to be done further

PratikMane0112 commented:
@viceice, I'm open for further fixtures & suggestions :)

viceice commented:
> @viceice, I'm open for further fixtures & suggestions :)

how's the code coverage now?
